zerver/tests/test_signup.py

```diff
@@ -3405,22 +3418,9 @@ def test_signup_invalid_name(self) -> None:
         """
         Check if an invalid name during signup is handled properly.
         """
-        email = "newguy@zulip.com"
-        password = "newpassword"
-
-        result = self.client_post("/accounts/home/", {"email": email})
-        self.assertEqual(result.status_code, 302)
-        self.assertTrue(result["Location"].endswith(f"/accounts/send_confirm/{email}"))
-        result = self.client_get(result["Location"])
-        self.assert_in_response("Check your email so we can get started.", result)
-
-        # Visit the confirmation link.
-        confirmation_url = self.get_confirmation_url_from_outbox(email)
-        result = self.client_get(confirmation_url)
-        self.assertEqual(result.status_code, 200)
 
-        # Pick a password and agree to the ToS.
-        result = self.submit_reg_form_for_user(email, password, full_name="<invalid>")
+        result = self.signup(full_name="<invalid>"
+                             stop_after_reg_form=True)
         self.assert_in_success_response(["Invalid characters in name!"], result)
```

Comment:
For this particular test, we're no longer checking the 200 status of the `result` on either end.  

I'd consider handling this issue by replacing `stop_after_reg_form` with `expect_registration_form_200_with_error`, and have the code path verify the 200 before returning the result.  I think it'd be reasonable to consider whether that parameter should be a boolean or the list of strings to look for in that error page; the latter might mean you don't need to inspect the returned `result` after all for most tests.

Reply:
In this case `self.assert_in_success_response` checks the 200 status of result. Do you still think your suggested approach is better?

Alternatively I could make the param `test_registration_form`, taking a function. If supplied call `test_registration_form`, passing in the response, if not do the default check. This might be overengineering.

Reply:
Yeah, I think so -- because we probably have (or should have!) N tests that expect a failure at this stage of registration.  So it'll be nice to have parameters noting where an error is expected, and `verify_signup` can just fork its implementation when that happens.